service.go

```diff
@@ -136,6 +136,7 @@ func GenerateRoute(s *v1.Service) *routev1.Route {
 			Labels:    labels,
 		},
 		Spec: routev1.RouteSpec{
+			Path: path
```

Comment:
This path gets auto-generated if its not included. We should handle the logic where the user provides it or not so they have the option of providing their own or letting OpenShift generate it since it has a predictable pattern .

Reply:
While trying with the marketplace operator, the path wasn't getting generated, because of which the metrics were not exposed. Do I make a default "/customMetrics" path, like we have "defaultMetricsPort" ?

Reply:
Ah sorry, I was confused when I wrote this. 

Let me retry. This path won't automatically be generated if left out. However, if the route is made over the service will have that endpoint available. If our route is generated with no path and the URL is
`myproject.mycluster.com`, since the service exposes a '/metrics endpoint, then `myproject.mycluster.com/metrics` would still work, even though the path is not part of the actual url from the route.